healthcheck/api/v1alpha1/healthcheck_types.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have
+// json tags for the fields to be serialized.
+// HealthCheckSpec defines the metadata of a single health check.
+type HealthCheckSpec struct {
+}
+
+// HealthCheckStatus defines the status of a single health check.
+type HealthCheckStatus struct {
+	// Conditions represents the status of health check.
+	// +kubebuilder:validation:MaxItems=1
+	Conditions []HealthCheckCondition `json:"conditions,omitempty"`
+}
+
+// HealthCheckConditionType defines the type of health check conditions.
+type HealthCheckConditionType string
+
+// The valid conditions of health check.
+const (
+	FatalError    HealthCheckConditionType = "FatalError"
+	NonFatalError HealthCheckConditionType = "NonFatalError"
+)
+
+// HealthCheckCondition represents the status of health check.
+type HealthCheckCondition struct {
+	// +kubebuilder:validation:Enum=FatalError;NonFatalError
+	Type HealthCheckConditionType `json:"type,omitempty"`
+	// +kubebuilder:validation:Enum=Unknown;Healthy;Unhealthy
+	Status metav1.ConditionStatus `json:"status,omitempty"`
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty"`
+	// The unique error ID.
+	CanonicalID string `json:"canonicalID,omitempty"`
+	// The unique error name.
+	CanonicalName string `json:"canonicalName,omitempty"`
```

Comment:
Is this the same as `Reason` ?

Reply:
The CanonicalName is the name for a type of health check error. The val extends the meaning of the CannoicalID. ex https://cloud.google.com/anthos-config-management/docs/reference/errors#knv1000

The Reason field is a set of predefined events that explains the reason for the last transition. ex: https://cloud.google.com/config-connector/docs/how-to/monitoring-your-resources#-specific_events. IIUC, they are more general?